### PR TITLE
SONARTFVC-22 Adding support for Visual Studio 2015

### PR DESCRIPTION
**Pull request by salehahu-msft**
Enabling support for Visual Studio 2015.
Will enable support for tfs 2015 in upcoming changes.

### CODE REVIEW

**Comment by demuruge** [comment]
CI failing, as we are taking dependency on vs 2015 - tfs (2015) dlls at compile time.

**Comment by dbolkensteyn** [comment]
@demuruge How do you want to solve this? I'd see 2 options
1. Add a NuGet dependency on the TFS dll and restore them during the build - https://www.nuget.org/packages/Microsoft.TeamFoundationServer.Client/
2. Use a Dev14 environment on AppVeyor

**Comment by demuruge** [comment]
Can we use the latest VS environment on AppVeyor? Can you let us know how we can add this?

**Comment by demuruge** [comment]
@dbolkensteyn Can you let us know how we can install dev14 in the App Veyor CI agent. I guess I don't have permission to do this.

**Comment by dbolkensteyn** [comment]
In fact you do, all the config is in the appveyor.yml file within this repository which you can change.
However I can give it a try tomorrow.

**Comment by dbolkensteyn** [comment]
@demuruge I've just pushed the change into the master - I'll relaunch this PR build to see if that helped

File: SonarTfsAnnotateTest/pom.xml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            </configuration>
+          </execution>
+          <execution>
+            <id>mstest</id>
+            <phase>test</phase>
+            <goals>
+              <goal>exec</goal>
+            </goals>
+            <configuration>
+              <executable>${env.MSTEST_PATH}</executable>
```
Comment (dbolkensteyn) [inline SonarTfsAnnotateTest/pom.xml]: This variable isn't defined, hence the PR is currently failing with:

```
 1109[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.3.2:exec (mstest) on project SonarTfsAnnotateTest: The parameters 'executable' for goal org.codehaus.mojo:exec-maven-plugin:1.3.2:exec are missing or invalid -> [Help 1]
```

**Comment by dbolkensteyn** [comment]
And it's green again :)

FYI (just my 2 cents) - the NuGet approach as in the other PR https://github.com/SonarCommunity/sonar-scm-tfvc/pull/11 seems to be like a good idea, because you'll redistribute the required assemblies together with the plugin (so there won't be a need to manually install the TFS 2015 Object Model on build agents) - but feel free to chose the approach you think is best.

**Pull request closed without merging**